swagger/swagger.yaml

```diff
@@ -3,23 +3,180 @@ info:
+              schema:
+                $ref: "#/components/schemas/Error"
+  /listings/{_id}:
+    get:
+      summary: Get a Listing by uuid
+      operationId: getListing
+      tags:
+        - Listings
+      parameters:
+        - name: _id
+          in: path
+          description: listing _id
+          required: true
+          schema:
+            type: string
+            format: string
+      responses:
+        "200":
+          description: The found Listing
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ResponseListing"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+    delete:
+      summary: Delete a Listing
+      operationId: deleteListing
+      tags:
+        - Listings
+      parameters:
+        - name: _id
+          in: path
+          description: listing _id
+          required: true
+          schema:
+            type: string
+            format: string
+      responses:
+        "200":
+          description: The deleted Listing
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ResponseListing"
+        "204":
+          description: No change (Nothing was deleted)
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+    put:
+      summary: Update a Listing
+      operationId: updateListing
+      tags:
+        - Listings
+      parameters:
+        - name: _id
+          in: path
+          description: listing _id
+          required: true
+          schema:
+            type: string
+            format: string
+      requestBody:
+        description: the listing to update
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Listing"
+      responses:
+        "201":
+          description: The created Listing
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ResponseListing"
+        "200":
+          description: The updated Listing
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ResponseListing"
+        "204":
+          description: No change (Nothing was deleted)
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+  /listings/?name={location}:
+    get:
+      summary: Get all Listing from Location
+      operationId: getListingbyLocation
+      tags:
+        - Listings
+      parameters:
+        - name: location
+          in: path
```

Comment:
`in: query` och `required: false` räcker bra för en queryparameter bara flytta den parametern till get /listing så är du klar